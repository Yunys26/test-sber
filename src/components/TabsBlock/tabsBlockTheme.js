
const { makeStyles, createMuiTheme } = require("@material-ui/core");

export const themeTabs = createMuiTheme({
    overrides: {
        MuiTab: {
            root: {
                fontSize: '20px',
            }
        },
        MuiCssBaseline: {
            '@global': {
                a: {
                    fontSize: '20px',
                    fontWeight: 'bold',
                    color: '#f50057',
                    textDecoration: 'none',
                    '&:hover': {
                        color: 'black',
                    }
                },
            },
        },
    },
})

export const useTabsBlockStyle = makeStyles({
    nameCompany: {
        marginBottom: "30px",
        fontSize: "41px",
        padding: "10px",
        backgroundColor: "black",
        borderRadius: "20px",
        color: "white",
        '&:hover': {
            transition: "0.5s",
            backgroundColor: "white",
            color: "Black",
        }
    },
    workBlock: {
        borderRadius: "20px",
        backgroundColor: "#e8e8e8",
        alignItems: "center",
        padding: "40px",
    },
    workBlockLogo: {
        width: "400px",
        height: "160px",
    },
    likeBlack: {
        width: "64px",
        height: "64px",
        '&:active': {
            color: "#f50057",
        },
    },
    likePrimary: {
        width: "64px",
        height: "64px",
        '&:active': {
            color: "#f50057",
        },
    },
    blockContent: {
        marginBottom: "40px",
    },
    headerSearchBlock: {
        backgroundColor: "black",
        color: "white",
        borderRadius: "20px",
        textAlign: "center",
        marginBottom: "20px",
    },
    titleSearchBlock: {
        marginBottom: "50px"
    },
    footerSearchBlock: {
        borderRadius: "20px",
        marginTop: "20px",
        backgroundColor: "black",
        color: "#f50057",
        textAlign: "center",
    },
    paperTabs: {
        padding: '5px',
        backgroundColor: '#202020',
        borderRadius: '20px',
        marginBottom: '20px',
        border: 'solid 1px black',
        color: 'white',
        transition: '0.5s',
        '&:hover': {
            transition: "0.8s",
            backgroundColor: "#f50057",
            color: "black",
        },
    },
    tabNavLink: {
        textDecoration: 'none',
        color: 'white',
        '&:hover': {
            color: '#202020',
        },
    },
    textTitleTabs: {
        textAlign: 'center',
    },
    statusLoadingText: {
        fontWeight: 'bold',
        marginTop: '20px',
        color: '#f50057',
    }
});