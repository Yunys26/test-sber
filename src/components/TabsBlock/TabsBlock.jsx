import React, { useState } from 'react';
// Libs
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, Route } from 'react-router-dom';
import {
    Tabs,
    Tab,
    Paper,
    ThemeProvider,
    Typography
} from '@material-ui/core';
// Theme
import { themeTabs, useTabsBlockStyle } from './tabsBlockTheme';
// Store action
import { showListLocalStorage, deleteWorkInLocalStorageStore } from '../../store/slicesStore/mainSlice';
// Components
import ListWork from '../List/ListWork';
import ModalS from '../Modal/Modal';
import ListFavorite from '../List/ListFavorite';
    

export default function TabsBlock(props) {

    const { inputValue, setInputValue } = props;

    const favoritesWorkList = useSelector(state => state.main.localStorageStore);

    const modalStateStore = useSelector(state => state.main.modalState);

    const data = useSelector(state => state.main.dataResponse);

    const classes = useTabsBlockStyle();

    const dispatch = useDispatch();

    const [valueIndexOne, setValueIndexOne] = useState(0);
    console.log(data)
    // Добавление/Удаление в избранное в tab work
    const addAndDelInFavorite = (index) => {
        for (let i = 0; i < localStorage.length; i++) {
            if (localStorage.key(i) === `${inputValue}?${index}`) {
                return localStorage.removeItem(localStorage.key(i));
            }
        }
        localStorage.setItem(`${inputValue}?${index}`, JSON.stringify(data[index]));
    };

    // Копирование localStorage в массив для хранения в хранилище 
    const showLocalStorage = () => {
        let arrayLocalStorage = [];
        if (localStorage.length !== 0) {
            for (let i = 0; i < localStorage.length; i++) {
                arrayLocalStorage.push(JSON.parse(localStorage.getItem(localStorage.key(i))));
            }
        }
        return arrayLocalStorage;
    };
    
    // Удаление из localStorage и хранилища 
    const deleteStoreAndLocalStorage = (index) => {
        localStorage.removeItem(localStorage.key(index));
        dispatch(deleteWorkInLocalStorageStore(favoritesWorkList.filter(data => data.id !== favoritesWorkList[index].id)));
    };

    // Логика работы Tabs
    const tabsIndexOne = () => {
        (valueIndexOne === 0 && setValueIndexOne(1)) || (valueIndexOne === 1 && setValueIndexOne(0));
    };

    return (
        <>
            <Paper className={classes.paperTabs} square>
                <Tabs
                    indicatorColor="secondary"
                    textColor="secondary"
                    centered
                    onClick={() => tabsIndexOne()}
                    value={valueIndexOne}
                >
                    <ThemeProvider theme={themeTabs}>
                        <NavLink className={classes.tabNavLink} to="/">
                            <Tab label="Work" value={0}/>
                        </NavLink>
                    </ThemeProvider>
                    <NavLink 
                        onClick={() => {
                            dispatch(showListLocalStorage(showLocalStorage()));
                            setInputValue('');
                        }}
                        className={classes.tabNavLink} to="/work"
                    >
                        <Tab label="Work Favorites" value={1}/>
                    </NavLink>
                </Tabs>
            </Paper>
            <Route exact path="/">
                {data.length === 1 && data.map( (el) => <Typography className={classes.statusLoadingText} variant="h2" align="center">{el.status}</Typography>) ||
                    <ListWork
                        classes={classes}
                        data={data}
                        funcLogic={addAndDelInFavorite}
                        statusList={<Typography variant="h4" align="center">Work list is empty</Typography>}
                    />
                }
            </Route>
            <Route path="/work">
                {data.length === 1 && data.map( (el) => <Typography className={classes.statusLoadingText} variant="h2" align="center">{el.status}</Typography>) ||
                    <ListFavorite
                        classes={classes}
                        data={favoritesWorkList}
                        funcLogic={deleteStoreAndLocalStorage}
                        statusListFavorit={<Typography variant="h4" align="center">Work favorites list is empty</Typography>}
                    />
                }
            </Route>
            
            {modalStateStore && <ModalS open={modalStateStore} />}
        </>
    )
}

TabsBlock.propTypes = {
    inputValue: PropTypes.string,
    setInputValue: PropTypes.func,
}