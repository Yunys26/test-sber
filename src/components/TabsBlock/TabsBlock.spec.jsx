import React from 'react';
import { shallow } from 'enzyme';
import TabsBlock from './TabsBlock';
import { NavLink } from 'react-router-dom';
import { Tabs } from '@material-ui/core';

// Мокаем useDispatch
jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn(),
    useSelector: () => jest.fn(),
}));

describe('<TabsBlock />', () => {
    let props, componentTabsBlock;

    beforeEach(() => {

        props = {
            data: [],
            setInputValue: (state) => state
        }

        componentTabsBlock = shallow(<TabsBlock {...props} />)

    })

    it('Проверка на рендер компонента', () => {
        expect(componentTabsBlock.length).toBe(1);
    });

    it('Snapshot TabsBlock', () => {
        expect(componentTabsBlock).toMatchSnapshot();
    });

    it('Проверяем кол-во NavLink', () => {
        expect(componentTabsBlock.find(NavLink)).toHaveLength(2);
    })

    it('Проверка клика на 2-ой NavLink', () => {
        const NavLinkFavorite = componentTabsBlock.find(NavLink).at(1);
        NavLinkFavorite.simulate('click', props.setInputValue(''));
    });

    it('Проврка на наличие Tabs', () => {
        expect(componentTabsBlock.find(Tabs)).toHaveLength(1);
    });

    it('Проверка клика Tabs', () => {
        const TabsClick = componentTabsBlock.find(Tabs);
        TabsClick.simulate('click', () => {})
    });

});