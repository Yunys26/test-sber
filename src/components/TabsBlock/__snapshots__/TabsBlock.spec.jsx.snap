// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsBlock /> Snapshot TabsBlock 1`] = `
<div>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paperTabs-10"
    square={true}
  >
    <WithStyles(ForwardRef(Tabs))
      centered={true}
      onClick={[Function]}
      value={0}
    >
      <NavLink
        className="makeStyles-tabNavLink-11"
        to="/"
      >
        <WithStyles(ForwardRef(Tab))
          label="Work"
          value={0}
        />
      </NavLink>
      <NavLink
        className="makeStyles-tabNavLink-11"
        onClick={[Function]}
        to="/work"
      >
        <WithStyles(ForwardRef(Tab))
          label="Work Favorites"
          value={1}
        />
      </NavLink>
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Paper))>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <ListWork
        classes={
          Object {
            "blockContent": "makeStyles-blockContent-6",
            "footerSearchBlock": "makeStyles-footerSearchBlock-9",
            "headerSearchBlock": "makeStyles-headerSearchBlock-7",
            "likeBlack": "makeStyles-likeBlack-4",
            "likePrimary": "makeStyles-likePrimary-5",
            "nameCompany": "makeStyles-nameCompany-1",
            "paperTabs": "makeStyles-paperTabs-10",
            "statusLoadingText": "makeStyles-statusLoadingText-13",
            "tabNavLink": "makeStyles-tabNavLink-11",
            "textTitleTabs": "makeStyles-textTitleTabs-12",
            "titleSearchBlock": "makeStyles-titleSearchBlock-8",
            "workBlock": "makeStyles-workBlock-2",
            "workBlockLogo": "makeStyles-workBlockLogo-3",
          }
        }
        data={[MockFunction]}
        local={[MockFunction]}
        showDataText={[Function]}
        statusList={
          <ForwardRef(WithStyles)
            align="center"
            variant="h4"
          >
            Work list is empty
          </ForwardRef(WithStyles)>
        }
      />
    </Route>
    <Route
      path="/work"
    >
      <ListFavorite
        classes={
          Object {
            "blockContent": "makeStyles-blockContent-6",
            "footerSearchBlock": "makeStyles-footerSearchBlock-9",
            "headerSearchBlock": "makeStyles-headerSearchBlock-7",
            "likeBlack": "makeStyles-likeBlack-4",
            "likePrimary": "makeStyles-likePrimary-5",
            "nameCompany": "makeStyles-nameCompany-1",
            "paperTabs": "makeStyles-paperTabs-10",
            "statusLoadingText": "makeStyles-statusLoadingText-13",
            "tabNavLink": "makeStyles-tabNavLink-11",
            "textTitleTabs": "makeStyles-textTitleTabs-12",
            "titleSearchBlock": "makeStyles-titleSearchBlock-8",
            "workBlock": "makeStyles-workBlock-2",
            "workBlockLogo": "makeStyles-workBlockLogo-3",
          }
        }
        local={[MockFunction]}
        showDataText={[Function]}
        statusListFavorit={
          <ForwardRef(WithStyles)
            align="center"
            variant="h4"
          >
            Work favorites list is empty
          </ForwardRef(WithStyles)>
        }
      />
    </Route>
  </Switch>
  <ModalS
    open={[MockFunction]}
  />
</div>
`;
